var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -117,6 +117,22 @@ class Hdf5(AutotoolsPackage):
     patch('h5public-skip-mpicxx.patch', when='@:1.8.21,1.10.0:1.10.5+mpi~cxx',
           sha256='b61e2f058964ad85be6ee5ecea10080bf79e73f83ff88d1fa4b602d00209da9c')
 
+    # The argument 'buf_size' of the C function 'h5fget_file_image_c' is
+    # declared as intent(in) though it is modified by the invocation. As a
+    # result, aggressive compilers such as Fujitsu's may do a wrong
+    # optimization to cause an error.
+    def patch(self):
+        filter_file(
+            'INTEGER(SIZE_T), INTENT(IN) :: buf_size',
+            'INTEGER(SIZE_T), INTENT(OUT) :: buf_size',
```

Comment:
As far as I remember `INTENT(OUT)` is a hint to the compiler that the value only matters as output and can thus be unspecified (e.g. uninitialized) when entering the subroutine. Didn't look at the code, but since the variable was declared `INTENT(IN)` has anyone checked if `INTENT(INOUT)` should be specified instead?

Reply:
@alalazo You can find `h5fget_file_image_c` in `fortran/src/H5Ff.c`, where its fourth argument `buf_req` isn't referenced but just updated. That is, it should be specified as `intent(out)`.

```
int_f
h5fget_file_image_c(hid_t_f *file_id, void *buf_ptr, size_t_f *buf_len, size_t_f *buf_req)
/******/
{
    herr_t ret_value=0;          /* Return value */
    ssize_t c_buf_req;
    /*
     * Call h5fget_file_image function
     */
    
    if ( (c_buf_req = H5Fget_file_image((hid_t)*file_id, buf_ptr, (size_t)*buf_len)) < 0)
         HGOTO_DONE(FAIL);

    *buf_req = (size_t_f)c_buf_req;

done:
      return ret_value;
}
```